### PR TITLE
feat: proto definitions for iceberg orders

### LINKED ISSUE #8359: [Feature]: Iceberg orders - proto definitions
# Feature Overview

**In order to** expose iceberg orders to users of vega
**We will** extend the proto definitions
**So that** fields relating to iceberg orders can be set when submitting an order

## Specs
- [SLA spec](https://github.com/vegaprotocol/specs/pull/1685) to spec or section within a spec

### PR DESCRIPTION
**Pull request by wwestgarth**
closes #8359 

The protos definitions for iceberg orders including all the to/from proto gubbins, and iceberg order-submission validation.

### CODE REVIEW

**Review by jeremyletang — DISMISSED** [review]

File: protos/sources/vega/commands/v1/commands.proto [hunk 1]
```diff
@@ -56,6 +56,16 @@ message OrderSubmission {
...
   bool reduce_only = 11;
+  // Parameters used to specify an iceberg order.
+  optional IcebergOpts iceberg_opts = 12;
+}
+
+// Iceberg order options
+message IcebergOpts {
+  // Size of the order that is initially made visible and can cause a trade within in a single transaction.
+  uint64 initial_peak_size = 1;
+  // Threshold at which the order's visible remaining will be refreshed back its initial peak size.
```
Comment (candida-d) [inline protos/sources/vega/commands/v1/commands.proto:67]:
```suggestion
  // Threshold at which the order's visible remaining will be refreshed back its initial peak size.
```
remaining size?

File: protos/sources/vega/vega.proto [hunk 2]
```diff
@@ -108,6 +108,16 @@ message PeggedOrder {
   string offset = 2;
 }
 
+// Details of an iceberg order
+message IcebergOrder {
+  // Size of the order that will be made visible if the it is refreshed at the end of a transaction.
```
Comment (candida-d) [inline protos/sources/vega/vega.proto:113]:
```suggestion
  // Size of the order that will be made visible if the iceberg order is refreshed at the end of a transaction.
```

File: protos/sources/vega/vega.proto [hunk 3]
```diff
@@ -108,6 +108,16 @@ message PeggedOrder {
...
+  uint64 initial_peak_size = 1;
+  // Threshold at which the order's visible remaining will be refreshed back its initial peak size.
```
Comment (candida-d) [inline protos/sources/vega/vega.proto:115]: remaining size question here too

File: protos/sources/vega/commands/v1/commands.proto [hunk 4]
```diff
@@ -56,6 +56,16 @@ message OrderSubmission {
...
   bool reduce_only = 11;
+  // Parameters used to specify an iceberg order.
+  optional IcebergOpts iceberg_opts = 12;
+}
+
+// Iceberg order options
+message IcebergOpts {
+  // Size of the order that is initially made visible and can cause a trade within in a single transaction.
+  uint64 initial_peak_size = 1;
+  // Threshold at which the order's visible remaining size will be refreshed back its initial peak size.
```
Comment (candida-d) [inline protos/sources/vega/commands/v1/commands.proto:67]:
```suggestion
  // Threshold at which the order's visible remaining size will be refreshed back to its initial peak size.
```
Sorry - missed this one in the last review

File: protos/sources/vega/vega.proto [hunk 5]
```diff
@@ -108,6 +108,16 @@ message PeggedOrder {
   string offset = 2;
 }
 
+// Details of an iceberg order
+message IcebergOrder {
+  // Size of the order that will be made visible if the iceberg order is refreshed at the end of a transaction.
+  uint64 initial_peak_size = 1;
+  // Threshold at which the order's visible remaining size will be refreshed back its initial peak size.
```
Comment (candida-d) [inline protos/sources/vega/vega.proto:115]:
```suggestion
  // Threshold at which the order's visible remaining size will be refreshed back to its initial peak size.
```

**Review by jeremyletang — DISMISSED** [review]

**Review by candida-d — APPROVED** [review]

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**